<ui version="4.0" >
 <class>Config</class>
 <widget class="QWidget" name="Config" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>497</width>
    <height>512</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Configure Text Analyser</string>
  </property>
  <property name="toolTip" >
   <string/>
  </property>
  <property name="whatsThis" >
   <string/>
  </property>
  <property name="sizeGripEnabled" stdset="0" >
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" >
   <item>
    <widget class="QTabWidget" name="tabWidget2" >
     <property name="currentIndex" >
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>Content</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5" >
       <item>
        <widget class="QGroupBox" name="groupBox" >
         <property name="title" >
          <string>Determine Language</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout" >
          <item>
           <widget class="QCheckBox" name="StaticLang" >
            <property name="toolTip" >
             <string>Set if the languages are static.</string>
            </property>
            <property name="whatsThis" >
             <string>When the languages are static and when the content of a document is changed, the system supposes that the language of this document has not changed.&lt;br>
When the languages are not static, the language of a document is always determinewhen the document is analysed.</string>
            </property>
            <property name="text" >
             <string>Static Languages [StaticLang]</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel1" >
              <property name="text" >
               <string>Minimum stopword ratio [MinStopWords]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer2" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KDoubleNumInput" name="MinStopWords" >
              <property name="toolTip" >
               <string>Set the minimum stopword ratio.</string>
              </property>
              <property name="whatsThis" >
               <string>To determine the language of a document, the following method is used:
&lt;ol>
&lt;li> For each language defined in the system, the ratio of the number of different stopwords over the overall number of different words contained in the document is computed.&lt;/li>
&lt;li> All the languages having this ratio lower than the &lt;b>minimum stopword ratio&lt;/b> are elminated.&lt;/li>
&lt;li> The language with the highest number of occurences of stopwords is assigned to the document.&lt;li>
&lt;/ol></string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2" >
         <property name="title" >
          <string>Valid Words</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2" >
          <item>
           <widget class="QCheckBox" name="NonLetterWords" >
            <property name="toolTip" >
             <string>Should non letter words be used.</string>
            </property>
            <property name="whatsThis" >
             <string>When analysing a document, it is possible to use for the description only words containing letters.</string>
            </property>
            <property name="text" >
             <string>Non letter words are used [NonLetterWords]</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel2" >
              <property name="text" >
               <string>Minimum size of a word [MinWordSize]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer3" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="MinWordSize" >
              <property name="toolTip" >
               <string>Specify the minimum size of words.</string>
              </property>
              <property name="whatsThis" >
               <string>When a word is contained in the document and its size is lower than this threshold, the word is not used.</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel3" >
              <property name="text" >
               <string>Minimum size of a stem [MinStemSize]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer4" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="MinStemSize" >
              <property name="toolTip" >
               <string>Set the minimum size of stems.</string>
              </property>
              <property name="whatsThis" >
               <string>On all the words in a document containing only letters, a stemming algorithm is applied. When the resulting stem has a size lower than this threshold, the stem is not used.</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupFiltering" >
         <property name="title" >
          <string>Filtering</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3" >
          <item>
           <widget class="QCheckBox" name="Filtering" >
            <property name="text" >
             <string>Use Word Filtering?</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel1_2" >
              <property name="toolTip" >
               <string>Maximum number of following occurences of same letter .</string>
              </property>
              <property name="whatsThis" >
               <string>Maximum number of following occurences of same letter.</string>
              </property>
              <property name="text" >
               <string>Maximal occurences</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer6" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="NbSameOccur" >
              <property name="toolTip" >
               <string>of a same word in a row to consider a word non-valid</string>
              </property>
              <property name="whatsThis" >
               <string>of a same word in a row to consider a word non-valid</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel2_2" >
              <property name="toolTip" >
               <string>Value of the ratio of normal letters on the total number of letters to be considered as a valid.</string>
              </property>
              <property name="whatsThis" >
               <string>Value of the ratio of normal letters on the total number of letters to be considered as a valid.</string>
              </property>
              <property name="text" >
               <string>Normal Ratio</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer7" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KDoubleNumInput" name="NormalRatio" />
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="spacer11" >
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType" >
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>16</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="TabPage" >
      <attribute name="title" >
       <string>XML Structure</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4" >
       <item>
        <widget class="QGroupBox" name="groupBox7" >
         <property name="title" >
          <string>General</string>
         </property>
         <layout class="QVBoxLayout" >
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QCheckBox" name="UseDefaultNamespace" >
              <property name="text" >
               <string>Use default namespace</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="DefaultNamespace" />
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="ExtractStruct" >
            <property name="text" >
             <string>Extract tags and attributes [ExtractStruct]</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="FullIndex" >
            <property name="text" >
             <string>Full index (structure and positions) [FullIndex]</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox6" >
         <property name="title" >
          <string>Elements Considered As Content</string>
         </property>
         <layout class="QVBoxLayout" >
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QCheckBox" name="StructIsContent" >
              <property name="text" >
               <string>Structurral tokens [StructIsContent]</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer28" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>20</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KDoubleNumInput" name="WeightStruct" />
            </item>
            <item>
             <widget class="QLabel" name="textLabel3_2" >
              <property name="text" >
               <string>[WeightStruct]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer15_2" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QCheckBox" name="ExtractValues" >
              <property name="text" >
               <string>Attributes values [ExtractValues]</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer29" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>30</width>
                <height>21</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KDoubleNumInput" name="WeightValues" />
            </item>
            <item>
             <widget class="QLabel" name="textLabel4_2" >
              <property name="text" >
               <string>[WeightValues]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer16" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>16</width>
                <height>21</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="Indexes" >
         <property name="title" >
          <string>Indexes</string>
         </property>
         <layout class="QVBoxLayout" >
          <item>
           <widget class="QCheckBox" name="ExtractIndex" >
            <property name="text" >
             <string>Extract declarative tags? [ExtractIndex]</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="ChildTags" >
            <property name="text" >
             <string>Declarative tags may have child tags? [ChildTags]</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel1_3" >
              <property name="text" >
               <string>Maximum terms in a declarative tag [MaxTerms]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer12" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="MaxTerms" >
              <property name="minValue" stdset="0" >
               <number>0</number>
              </property>
              <property name="maxValue" stdset="0" >
               <number>50</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel2_3" >
              <property name="text" >
               <string>Maximum depth of a declarative tag [MaxDepth]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer13" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="MaxDepth" >
              <property name="minValue" stdset="0" >
               <number>0</number>
              </property>
              <property name="maxValue" stdset="0" >
               <number>30</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" >
            <item>
             <widget class="QLabel" name="textLabel1_5" >
              <property name="text" >
               <string>Maximal number of occurences [MaxOccurs]</string>
              </property>
              <property name="wordWrap" >
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="spacer15" >
              <property name="orientation" >
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType" >
               <enum>QSizePolicy::Expanding</enum>
              </property>
              <property name="sizeHint" stdset="0" >
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="KIntNumInput" name="MaxOccurs" />
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>KDoubleNumInput</class>
   <extends>QWidget</extends>
   <header>knuminput.h</header>
  </customwidget>
  <customwidget>
   <class>KIntNumInput</class>
   <extends>QWidget</extends>
   <header>knuminput.h</header>
  </customwidget>
 </customwidgets>
 <includes>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
  <include location="local" >knuminput.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>Filtering</sender>
   <signal>toggled(bool)</signal>
   <receiver>NbSameOccur</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>62</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel" >
     <x>111</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Filtering</sender>
   <signal>toggled(bool)</signal>
   <receiver>NormalRatio</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>62</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel" >
     <x>111</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractIndex</sender>
   <signal>toggled(bool)</signal>
   <receiver>ChildTags</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel" >
     <x>37</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractIndex</sender>
   <signal>toggled(bool)</signal>
   <receiver>ChildTags</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel" >
     <x>37</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractIndex</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaxDepth</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel" >
     <x>460</x>
     <y>446</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractIndex</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaxTerms</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel" >
     <x>459</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractStruct</sender>
   <signal>toggled(bool)</signal>
   <receiver>StructIsContent</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel" >
     <x>38</x>
     <y>226</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StructIsContent</sender>
   <signal>toggled(bool)</signal>
   <receiver>WeightStruct</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>38</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel" >
     <x>313</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractValues</sender>
   <signal>toggled(bool)</signal>
   <receiver>WeightValues</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>38</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel" >
     <x>306</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>UseDefaultNamespace</sender>
   <signal>toggled(bool)</signal>
   <receiver>DefaultNamespace</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>38</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel" >
     <x>210</x>
     <y>100</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ExtractIndex</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaxOccurs</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>37</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel" >
     <x>460</x>
     <y>487</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>toggleStruct(bool)</slot>
  <slot>toggleFullIndex(bool)</slot>
 </slots>
</ui>
